_stbt/transition.py

```diff
@@ -221,44 +222,48 @@ def _ddebug(s, f, *args):
-        f2 = frame[region.y:region.bottom, region.x:region.right]
-
-    absdiff = cv2.absdiff(f1, f2)
-    if mask is not None:
-        absdiff = cv2.bitwise_and(absdiff, mask, absdiff)
-
-    diffs_found = False
-    out_region = None
-    maxdiff = numpy.max(absdiff)
-    if maxdiff > 20:
-        diffs_found = True
-        big_diffs = absdiff > 20
-        out_region = pixel_bounding_box(big_diffs)
-        _ddebug("found %s diffs above 20 (max %s) in %r", frame,
-                numpy.count_nonzero(big_diffs), maxdiff, out_region)
-    elif maxdiff > 0:
-        small_diffs = absdiff > 5
-        small_diffs_count = numpy.count_nonzero(small_diffs)
-        if small_diffs_count > 50:
+class StrictDiff(FrameDiffer):
+    """The original `press_and_wait` algorithm."""
+
+    def diff(self, frame):
+        absdiff = cv2.absdiff(crop(self.prev_frame, self.region),
+                              crop(frame, self.region))
+        if self.mask is not None:
+            absdiff = cv2.bitwise_and(absdiff, self.mask, absdiff)
+
+        diffs_found = False
+        out_region = None
+        maxdiff = numpy.max(absdiff)
+        if maxdiff > 20:
             diffs_found = True
-            out_region = pixel_bounding_box(small_diffs)
-            _ddebug("found %s diffs <= %s in %r", frame, small_diffs_count,
-                    maxdiff, out_region)
-        else:
-            _ddebug("only found %s diffs <= %s", frame, small_diffs_count,
-                    maxdiff)
-    if out_region:
-        out_region = out_region.translate(region)
-
-    result = MotionResult(getattr(frame, "time", None), diffs_found,
-                          out_region, frame)
-    draw_on(frame, result, label="transition")
-    return result
+            big_diffs = absdiff > 20
+            out_region = pixel_bounding_box(big_diffs)
+            _ddebug("found %s diffs above 20 (max %s) in %r", frame,
+                    numpy.count_nonzero(big_diffs), maxdiff, out_region)
+        elif maxdiff > 0:
+            small_diffs = absdiff > 5
+            small_diffs_count = numpy.count_nonzero(small_diffs)
+            if small_diffs_count > 50:
+                diffs_found = True
+                out_region = pixel_bounding_box(small_diffs)
+                _ddebug("found %s diffs <= %s in %r", frame, small_diffs_count,
+                        maxdiff, out_region)
+            else:
+                _ddebug("only found %s diffs <= %s", frame, small_diffs_count,
+                        maxdiff)
+        if diffs_found:
+            # Undo crop:
+            out_region = out_region.translate(self.region)
+            # Only update the reference frame if we found differences. This
+            # makes the algorithm more sensitive to slow motion.
+            self.prev_frame = frame
+
+        result = MotionResult(getattr(frame, "time", None), diffs_found,
+                              out_region, frame)
+        return result
+
+
+press_and_wait.differ = StrictDiff
```

Comment:
Yeah, it isn't intended to be used commonly. Either set it once to keep the old behaviour, or not at all. I'm wary of adding parameters to `press_and_wait` because then we have to add them to all the callers, too (such as `Keyboard`).